Jose/Jwt.hs

```diff
@@ -60,8 +61,11 @@ encode :: MonadRandom m
 encode jwks encoding msg = runExceptT $ case encoding of
     JwsEncoding None -> case msg of
         Claims p -> return $ Jwt $ BC.intercalate "." [unsecuredHdr, B64.encode p]
-        Nested _ -> throwE BadClaims
-    JwsEncoding a    -> case filter (canEncodeJws a) jwks of
+        Nested Jwt {unJwt = p} ->
+          throwE
+          . BadClaims
+          $ "Encoding failed. Payload: " <> BC.unpack  p
```

Comment:
This should probably just be something like:

```
Nested _ -> throwE (BadClaims "A JWS can't contain a nested JWT")
```
since that's why the error is being reported. It would be a coding error rather than a client issue so not something that should happen in practice. I don't think there's a need to log the encode content.